specs/overview.md

```diff
@@ -0,0 +1,158 @@
+
+The below diagram illustrates how the sequencer and verifiers fit together:
+
+![Propagation](./assets/propagation.svg)
+
+## Key Interactions In Depth
+
+### Deposits
+
+**Spec links:**
+
+- [Deposits](./deposits.md)
+
+Optimism supports two types of deposits: user deposits, and L1 attributes deposits. To perform a user deposit, users call the `depositTransaction` method on the `DepositFeed` contract. This in turn emits `TransactionDeposited` events, which the rollup node reads during block derivation.
+
+L1 attributes deposits are used to register L1 block attributes (number, timestamp, etc.) on L2 via a call to the L1 Attributes Predeploy. They cannot be initiated by users, and are instead added to L2 blocks automatically by the rollup node.
+
+Both deposit types are represented by a custom EIP-2718 transaction type on L2.
+
+### Block Derivation
+
+#### Overview
+
+The rollup chain can be deterministically derived given an L1 Ethereum chain. The fact that the entire rollup chain can be derived based on L1 blocks is _what makes OE a rollup_. This process can be represented as:
+
+```
+derive_rollup_chain(l1_blockchain) -> rollup_blockchain
+```
+
+Optimism's block derivation function is designed such that it:
+
+- Requires no state other than what is easily accessible using L1 and L2 execution engine APIs.
+- Supports sequencers and sequencer consensus.
+- Is resilient to sequencer censorship.
+
+#### Epochs and the Sequencing Window
+
+The rollup chain is subdivided into epochs. There is a 1:1 correspondence between L1 block numbers and epoch numbers. For L1 block number `n`, there is a corresponding rollup epoch `n` which can only be derived after L1 block number `n` is added to the L1 chain. 
+
+Each epoch contains at least one block. Every block in the epoch contains an L1 info transaction which contains contextual about L1 such as the block hash and timestamp. The first block in the epoch also contains all deposits initiated via the `DepositFeed` contract on L1. All subsequent blocks contain batches of transactions submitted directly to the sequencer.
+
+The sequencer can create blocks for future epochs, and reference them before confirmation on L1. This is what enables _fast transaction confirmations_ on the sequencer. To prevent censorship, however, the sequencer must batch-submit the blocks belonging to each epoch within a fixed number of L1 blocks. This submission window is called the _sequencing window_. The `epoch`s described above correspond to the first block in each sequencing window. Since transaction batches for a given epoch can be submitted anywhere within the sequencing window, verifiers must search all blocks within the window for transaction batches.
+
+The following diagram describes this relationship, and how L2 blocks are derived from L1 blocks (L1 info transactions have been elided):
```

Comment:
Diagram looks good. Only comment is that with a larger sequencing window, batches can be in any order and more than one block later.

Reply:
I think this will never be my favorite diagram, but it seems correct!

Two notes though:
- Currently epochs will almost always have a fixed size (i.e. time between two L1 blocks / 2, so usually 6 blocks per epoch).
- If we adopt the variable-size epoch proposal (cf. "timestamp"  discussion) then it becomes possible to make bigger epochs and hence delay the first L2 block of an epoch much more. So if you're diagram is supposed to be "time-aligned" then the L1 block and first L2 block of the corresponding epoch won't necessarily line up so well anymore.